tests/rptest/tests/cloud_retention_test.py

```diff
@@ -8,60 +8,84 @@
 # by the Apache License, Version 2.0
 
 from ducktape.utils.util import wait_until
+from ducktape.mark import matrix
 from rptest.services.cluster import cluster
 from rptest.tests.prealloc_nodes import PreallocNodesTest
 from rptest.clients.types import TopicSpec
 from rptest.clients.rpk import RpkTool
 from rptest.services.redpanda import SISettings, MetricsEndpoint
 from rptest.services.kgo_verifier_services import (
     KgoVerifierConsumerGroupConsumer, KgoVerifierProducer)
+from rptest.utils.mode_checks import skip_debug_mode
 
 
 class CloudRetentionTest(PreallocNodesTest):
-    segment_size = 1024 * 1024
     default_retention_segments = 2
     topic_name = "si_test_topic"
 
     def __init__(self, test_context):
+        super(CloudRetentionTest, self).__init__(test_context=test_context,
+                                                 node_prealloc_count=1,
+                                                 num_brokers=3)
+
+    def setUp(self):
+        # Defer cluster startup so we can tweak configs based on the type of
+        # test run.
+        pass
+
+    @cluster(num_nodes=4)
+    @matrix(max_consume_rate_mb=[20, None])
```

Comment:
@ztlpn while we're moving this around, can you please remind us what the rationale is for running the test once with a consumer rate limit, then once without?

Reply:
The idea was to allow the fast consumer to experience the transition between segments before starting over. But this works only if the fast consumer is faster than eviction (i.e. faster than produce rate). But I guess in reality the fast consumer is slower than the producer too, so as of now it doesn't make much sense and to get into the desired regime we'll need to make producers slower.